src/measure_theory/arithmetic.lean

```diff
@@ -0,0 +1,497 @@
+  has_measurable_div G :=
+{ measurable_const_div := λ c,
+    by { convert (measurable_inv.const_mul c), ext1, apply div_eq_mul_inv },
+  measurable_div_const := λ c,
+    by { convert (measurable_id.mul_const c⁻¹), ext1, apply div_eq_mul_inv } }
+
+section inv
+
+variables {G : Type*} [has_inv G] [measurable_space G] [has_measurable_inv G]
+
+@[to_additive] lemma measurable.inv {f : α → G} (hf : measurable f) :
+  measurable (λ x, (f x)⁻¹) :=
+measurable_inv.comp hf
+
+@[to_additive] lemma ae_measurable.inv {f : α → G} {μ : measure α} (hf : ae_measurable f μ) :
+  ae_measurable (λ x, (f x)⁻¹) μ :=
+measurable_inv.comp_ae_measurable hf
+
+@[simp, to_additive] lemma measurable_inv_iff {G : Type*} [group G] [measurable_space G]
+  [has_measurable_inv G] {f : α → G} : measurable (λ x, (f x)⁻¹) ↔ measurable f :=
+⟨λ h, by simpa only [inv_inv] using h.inv, λ h, h.inv⟩
+
+@[simp, to_additive] lemma ae_measurable_inv_iff {G : Type*} [group G] [measurable_space G]
+  [has_measurable_inv G] {f : α → G} {μ : measure α} :
+  ae_measurable (λ x, (f x)⁻¹) μ ↔ ae_measurable f μ :=
+⟨λ h, by simpa only [inv_inv] using h.inv, λ h, h.inv⟩
+
+@[simp] lemma measurable_inv_iff' {G₀ : Type*} [group_with_zero G₀]
+  [measurable_space G₀] [has_measurable_inv G₀] {f : α → G₀} :
+  measurable (λ x, (f x)⁻¹) ↔ measurable f :=
+⟨λ h, by simpa only [inv_inv'] using h.inv, λ h, h.inv⟩
+
+@[simp] lemma ae_measurable_inv_iff' {G₀ : Type*} [group_with_zero G₀]
+  [measurable_space G₀] [has_measurable_inv G₀] {f : α → G₀} {μ : measure α} :
+  ae_measurable (λ x, (f x)⁻¹) μ ↔ ae_measurable f μ :=
+⟨λ h, by simpa only [inv_inv'] using h.inv, λ h, h.inv⟩
+
+end inv
+
+instance has_measurable_gpow (G : Type*) [group G] [measurable_space G]
+  [has_measurable_mul₂ G] [has_measurable_inv G] :
+  has_measurable_pow G ℤ :=
+by haveI : measurable_singleton_class ℤ := ⟨λ _, trivial⟩; exact
+⟨measurable_from_prod_encodable $
+  λ n, int.cases_on n measurable_id.pow_const (λ n, (measurable_id.pow_const (n + 1)).inv)⟩
+
+instance has_measurable_fpow (G₀ : Type*) [group_with_zero G₀] [measurable_space G₀]
+  [has_measurable_mul₂ G₀] [has_measurable_inv G₀] :
+  has_measurable_pow G₀ ℤ :=
+by haveI : measurable_singleton_class ℤ := ⟨λ _, trivial⟩; exact
+⟨measurable_from_prod_encodable $
+  λ n, int.cases_on n measurable_id.pow_const (λ n, (measurable_id.pow_const (n + 1)).inv)⟩
+
+@[priority 100, to_additive]
+instance has_measurable_div₂_of_mul_inv (G : Type*) [measurable_space G]
+  [div_inv_monoid G] [has_measurable_mul₂ G] [has_measurable_inv G] :
+  has_measurable_div₂ G :=
+⟨by { simp only [div_eq_mul_inv], exact measurable_fst.mul measurable_snd.inv }⟩
+
+/-- We say that the action of `M` on `α` `has_measurable_const_smul` if for each `c` the map
+`x ↦ c • x` is a measurable function. -/
+class has_measurable_const_smul (M α : Type*) [has_scalar M α] [measurable_space α] : Prop :=
+(measurable_const_smul : ∀ (c : M), measurable ((•) c : α → α))
```

Comment:
You are changing the design compared to the other operations. It would seem natural to require in `has_measurable_smul` that both operations are separately measurable, and in `has_measurable_smul₂` that the joint operation is measurable, as you do for the other operations. If you have a good reason to change the design here (in view of some particular applications), you should explain it in a docstring.

Reply:
The main reason is that we might want to talk about `measurable_const_smul` when `M` has no `measurable_space` instance.

Reply:
Do you have an example in mind? Otherwise, I'd rather have a more uniform (and less surprising) design.

Reply:
I moved to `has_measurable_smul`/`has_measurable_smul₂`. This breaks `ae_eq_fun.has_scalar`. Should I add `[measurable_space 𝕜] [opens_measurable_space 𝕜]`?

BTW, what about arithmetic operations on functional spaces without second countable topology? Is `(+)` usually a measurable operation w.r.t. the Borel σ-algebra?

Reply:
Done in 013e5e709 and 306df1baa